MBBSEmu/Logging/Targets/ConsoleTarget.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+namespace MBBSEmu.Logging.Targets
+{
+    /// <summary>
+    ///     Target for Logging Messages to the Console
+    /// </summary>
+    public class ConsoleTarget : ILoggingTarget
+    {
+        public ConsoleTarget() { }
+
+        public void Write(params object[] logEntry)
+        {
+            if(logEntry.Length != 0)
+                return;
+
+            var logMessage = (string)logEntry[0];
+            var logLevel = ((EnumLogLevel)logEntry[1]).ToString();
+
+            //Use reflection to get the name of the class calling this method
+            var callingType = new System.Diagnostics.StackTrace().GetFrame(3)?.GetMethod()?.DeclaringType;
```

Comment:
at least in Java, retrieving the stack trace is an expensive operation. If lots of logging occurs it'll bog down performance I'd imagine

Reply:
![image](https://github.com/mbbsemu/MBBSEmu/assets/1139047/866fba57-383c-4693-bf11-eb2d5791c376)

Reply:
I think in a refactor we can move the reflection of the calling type up the stack into the Logger. I imagine this is why you have to pass in `GetLogger<T>` in most logging libraries in order to save it from having to look up the calling type. Our release version doesn't log nearly as much where I think this will be of any material performance impact, but I'll mark it with a `TODO` so we don't lose track of it.